ql/time/asx.cpp

```diff
@@ -25,13 +25,16 @@
 #include <ql/settings.hpp>
 #include <ql/utilities/dataparsers.hpp>
 #include <boost/algorithm/string/case_conv.hpp>
+#include <boost/utility/string_view.hpp>
 #include <string>
-
-using boost::algorithm::to_upper_copy;
-using std::string;
+#include <cctype>
 
 namespace QuantLib {
 
+    namespace {
+        const boost::string_view All_MONTH_CODES = "FGHJKMNQUVXZ";
```

Comment:
We should move `string_view` to `ext` and enable the `std` version when `QL_USE_STD_CLASSES` is enabled, same as we did with `optional` and `any`.

Reply:
Should this be really done within this mr or rather a new issue/mr? I could try to implement that myself as well.

Reply:
I wouldn't bother as long as this is an implementation detail and not part of the interface.